server/service/osquery.go

```diff
@@ -1370,13 +1371,21 @@ func submitLogsEndpoint(ctx context.Context, request interface{}, svc fleet.Serv
 
 	case "result":
 		var results []json.RawMessage
-		if err := json.Unmarshal(req.Data, &results); err != nil {
+		if err = json.Unmarshal(req.Data, &results); err != nil {
 			err = newOsqueryError("unmarshalling result logs: " + err.Error())
 			break
 		}
-		err = svc.SubmitResultLogs(ctx, results)
-		if err != nil {
-			break
+
+		if errSave := svc.SaveResultLogsToQueryReports(ctx, results); errSave != nil {
+			combinedErrors = append(combinedErrors, errSave.Error())
+		}
```

Comment:
As discussed in Slack, let's not error out to osquery if the query results could not be stored, let's just log the error here.

Reply:
to be safe i removed all errs returned to osqueryd for query reports